### PR TITLE
[Reader] Fix post link deeplinks not loading properly

### LINKED ISSUE #20754: [Reader] Post Details not loaded from deep link
### Expected behavior
When tapping a WordPress.com site post link on a different app it should open the Reader feature of the Jetpack app in the Post Details screen, showing its content.

### Actual behavior
Tapping a WordPress.com site post link on a different app the first time opens the Reader in the Post Details with a message saying "The post no longer exists". Going back and tapping the **same** post link again works as expected.

### Steps to reproduce the behavior
1. On a different app (e.g.: Slack, Chrome, e-mail) that has a link to a post for a site hosted by WP.com
2. Tap a post link you never opened before (the site must match `*.wordpress.com` to open Jetpack)
3. The Jetpack app opens but shows the error message above
4. Go back/close the app
5. On the different app, tap the link again
6. The Jetpack app opens and shows the post content in Reader

### PR DESCRIPTION
**Pull request by thomashorta**
Fixes #20754

The issue was introduced in #20642, which just moved the local DB saving to a background thread, keeping the `requestListener` call outside. Having it outside the asynchronous thread execution block causes the callback to be called **BEFORE** the post is added to the table, which results in unexpected behavior, at least for posts opened via deep link.

This is fixed by moving the `requestListener` callback inside the background thread.

@antonis, can you also double-check the code changes here and test if everything still works as expected by the original PR expectations? I ran the test-steps from #20642 and things seem to be working fine.

-----

## To Test:

1. On a different app (e.g.: Slack, Chrome, e-mail) that has a link to a post for a site hosted by WP.com
2. Tap a post link you never opened before (the site must match `*.wordpress.com` to open Jetpack)
3. Verify the Jetpack app opens and shows the post content in the Reader

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - N/A

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
N/A

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/282">24.8 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by thomashorta** [comment]
@oguzkocer, this PR targets the `release/24.8` branch since the issue is a regression introduced in this version. We might want a new RC after this is merged, but I believe we can wait until Monday for that new release candidate since we might find out other regressions that require a fix, wdyt?

**Review by RenanLukas — APPROVED** [review]
Thanks for the fix, @thomashorta.
I went through the testing steps and it's working as expected. :shipit:

**Comment by antonis** [comment]
Thank you for the fix @thomashorta 🙇

**Pull request merged**
